cli/tools/registry/pm.rs

```diff
@@ -786,15 +781,34 @@ pub async fn remove(
       config.commit().await?;
     }
 
-    // Update deno.lock
-    node_resolver::PackageJsonThreadLocalCache::clear();
-    let cli_factory = CliFactory::from_flags(flags);
-    cache_deps::cache_top_level_deps(&cli_factory, None).await?;
+    npm_install_after_modification(flags, None).await?;
   }
 
   Ok(())
 }
 
+async fn npm_install_after_modification(
+  flags: Arc<Flags>,
+  // todo(dsherret): why is this dependency being passed here?
```

Comment:
Oops, that was me.

The reason is because we create our own `JsrFetchResolver` in `add`, and the file fetcher cache is tied to that resolver. If we create a new `JsrFetchResolver` in `cache_top_level_deps` instead of passing our existing one, it redownloads files (and logs again) because it has a new file fetcher cache.

It's very possible there's a cleaner way to fix that, though

Reply:
Ah I see. Thanks! I'll update this comment tomorrow.